builder/dockerfile/parser/parser.go

```diff
@@ -35,12 +37,29 @@ type Node struct {
 }
 
 var (
-	dispatch              map[string]func(string) (*Node, map[string]bool, error)
-	tokenWhitespace       = regexp.MustCompile(`[\t\v\f\r ]+`)
-	tokenLineContinuation = regexp.MustCompile(`\\[ \t]*$`)
-	tokenComment          = regexp.MustCompile(`^#.*$`)
+	dispatch                    map[string]func(string) (*Node, map[string]bool, error)
+	tokenWhitespace             = regexp.MustCompile(`[\t\v\f\r ]+`)
+	tokenLineContinuation       *regexp.Regexp
+	tokenEscape                 rune
+	tokenEscapeCommand          = regexp.MustCompile(`^#[ \t]*escape[ \t]*=[ \t]*(?P<escapechar>.).*$`)
+	tokenComment                = regexp.MustCompile(`^#.*$`)
+	builderInstructionProcessed bool
+	directiveEscapeSeen         bool
 )
 
+const defaultTokenEscape = "\\"
+
+// setTokenEscape sets the default token for escaping characters in a Dockerfile.
+func setTokenEscape(s string) error {
+	if s != "`" && s != "\\" {
+		return fmt.Errorf("invalid ESCAPE '%s'. Must be ` or \\", s)
```

Comment:
As put in the PR notes when submitting it, there's really no need to open this up to be any character at all at this time. If there's enough feedback/justification we could revisit that decision, then sure, let's consider that later. I mean, do you really want users to be able to do `# ESCAPE=e`? That would lead to all kinds of abuse and confusion when confronted with a line such as `COPY file c:\filepath'... It seems ` as the single new alternative to the default `\`, and being consistent with Powershell (read relatively natural to most Windows users), makes an awful lot of sense (to me).